### PR TITLE
Changed to version 3 for catalog roles

### PR DESCRIPTION
**Pull request by mkanoor**
- Operator should have been operation
- Added a new attribute filter for current username

### CODE REVIEW

File: rbac/management/role/definitions/catalog.json [hunk 1]
```diff
@@ -1,179 +1,278 @@
...
+  "roles": [
+    {
+      "name": "Catalog Administrator",
+      "system": true,
+      "version": 3,
+      "description": "A catalog administrator roles grants read, write and order permissions",
+      "access": [
+        {
+          "permission": "catalog:portfolios:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 2]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolios:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 3]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolios:order",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 4]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 5]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 6]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:order",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 7]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:orders:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 8]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:orders:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 9]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:orders:order",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 10]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:order_items:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 11]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "{{username}}",
+                "key": "owner",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:approval_requests:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 12]
```diff
@@ -1,179 +1,278 @@
...
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:order_items:order"
+        },
+        {
+          "permission": "catalog:progress_messages:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 13]
```diff
@@ -1,179 +1,278 @@
...
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:orders:order"
+        },
+        {
+          "permission": "catalog:order_items:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 14]
```diff
@@ -1,179 +1,278 @@
...
+        },
+        {
+          "permission": "catalog:portfolio_items:read"
+        },
+        {
+          "permission": "catalog:portfolio_items:order"
+        },
+        {
+          "permission": "catalog:orders:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 15]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:approval_requests:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 16]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:approval_requests:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 17]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:progress_messages:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 18]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:progress_messages:read",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 19]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:order_items:order",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 20]
```diff
@@ -1,179 +1,278 @@
...
+                "value": "*",
+                "key": "id",
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:order_items:write",
+          "resource_definitions": [
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
          "resourceDefinitions": [
```

File: rbac/management/role/definitions/catalog.json [hunk 21]
```diff
@@ -1,179 +1,278 @@
...
+      "name": "Catalog Administrator",
+      "system": true,
+      "version": 3,
+      "description": "A catalog administrator roles grants read, write and order permissions",
+      "access": [
+        {
+          "permission": "catalog:portfolios:read",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 22]
```diff
@@ -1,179 +1,278 @@
...
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolios:write",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 23]
```diff
@@ -1,179 +1,278 @@
...
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolios:order",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 24]
```diff
@@ -1,179 +1,278 @@
...
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:read",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 25]
```diff
@@ -1,179 +1,278 @@
...
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:write",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 26]
```diff
@@ -1,179 +1,278 @@
...
+                "operation": "equal"
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:portfolio_items:order",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
```
Comment (chambridge) [inline rbac/management/role/definitions/catalog.json]:
```suggestion
              "attributeFilter": {
```

File: rbac/management/role/definitions/catalog.json [hunk 27]
```diff
@@ -1,179 +1,278 @@
...
+              }
+            }
+          ]
+        },
+        {
+          "permission": "catalog:approval_requests:read",
+          "resource_definitions": [
+            {
+              "attribute_filter": {
+                "value": "{{username}}",
```
Comment (mkanoor) [inline rbac/management/role/definitions/catalog.json]: @chambridge I hope this doesn't get expanded when you store it in the database

Comment (chambridge) [reply rbac/management/role/definitions/catalog.json]: Looks fine when run locally. 
```
{
            "uuid": "0540e671-4eb6-4c03-840c-7a3f8f025788",
            "name": "Catalog User",
            "description": "A catalog user roles grants read and order permissions",
            "access": [
                {
                    "permission": "catalog:approval_requests:read",
                    "resourceDefinitions": [
                        {
                            "attributeFilter": {
                                "key": "owner",
                                "value": "{{username}}",
                                "operation": "equal"
                            }
                        }
                    ]
                },
                {
                    "permission": "catalog:portfolios:read",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:portfolios:order",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:portfolio_items:read",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:portfolio_items:order",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:orders:read",
                    "resourceDefinitions": [
                        {
                            "attributeFilter": {
                                "key": "owner",
                                "value": "{{username}}",
                                "operation": "equal"
                            }
                        }
                    ]
                },
                {
                    "permission": "catalog:orders:order",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:order_items:read",
                    "resourceDefinitions": [
                        {
                            "attributeFilter": {
                                "key": "owner",
                                "value": "{{username}}",
                                "operation": "equal"
                            }
                        }
                    ]
                },
                {
                    "permission": "catalog:order_items:order",
                    "resourceDefinitions": []
                },
                {
                    "permission": "catalog:progress_messages:read",
                    "resourceDefinitions": [
                        {
                            "attributeFilter": {
                                "key": "owner",
                                "value": "{{username}}",
                                "operation": "equal"
                            }
                        }
                    ]
                }
            ]
```

image

**Pull request merged**